packages/labs/cli/src/lib/lit-cli.ts

```diff
@@ -0,0 +1,151 @@
+    });
+  }
+
+  async run() {
+    const helpCommand = this.commands.get('help')!;
+    this.console.debug('running...');
+
+    // If the "--version" flag is ever present, just print
+    // the current version. Useful for globally installed CLIs.
+    if (this.args.indexOf('--version') > -1) {
+      const require = createRequire(import.meta.url);
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      console.log(require('../package.json').version);
+      return;
+    }
+
+    const result = this.getCommand(this.commands, this.args);
+    if ('invalidCommand' in result) {
+      return helpCommand.run({command: result.invalidCommand}, this.console);
+    } else {
+      const commandName = result.commandName;
+      const command = result.command;
+      const commandArgs = result.argv;
+
+      this.console.debug(
+        `command '${commandName}' found, parsing command args:`,
+        {args: commandArgs}
+      );
+
+      const commandDefinitions = mergeOptions([
+        command.options ?? [],
+        globalOptions,
+      ]);
+
+      const commandOptions = commandLineArgs(commandDefinitions, {
+        argv: commandArgs,
+      });
+      this.console.debug(`command options parsed from args:`, commandOptions);
+
+      // Help is a special argument for displaying help for the given command.
+      // If found, run the help command instead, with the given command name as
+      // an option.
+      if (commandOptions['help']) {
+        this.console.debug(
+          `'--help' option found, running 'help' for given command...`
+        );
+        return helpCommand.run({command: commandName}, this.console);
+      }
+
+      this.console.debug('Running command...');
+      return command.run(commandOptions, this.console);
+    }
+  }
+
+  getCommand(
+    commands: Map<string, Command>,
+    args: ReadonlyArray<string>,
+    parentCommandNames: Array<string> = []
+  ):
+    | {commandName: string; command: Command; argv: string[]}
+    | {invalidCommand: string} {
+    try {
+      const parsedArgs = commandLineCommands([...commands.keys()], [...args]);
+      const commandName = parsedArgs.command!;
+      const command = commands.get(commandName)!;
+      if (command.subcommands !== undefined && parsedArgs.argv.length > 0) {
+        const subcommands = new Map<string, Command>(
+          command.subcommands.map((c) => [c.name, c])
+        );
+        return this.getCommand(subcommands, parsedArgs.argv, [
+          ...parentCommandNames,
+          commandName,
+        ]);
+      }
+      return {
+        commandName: [...parentCommandNames, commandName].join(' '),
+        command,
+        argv: parsedArgs.argv,
+      };
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      // We need a valid command name to do anything. If the given
+      // command is invalid, run the generalized help command.
+      if (error.name === 'INVALID_COMMAND') {
```

Comment:
prefer an instanceof here, and for error to be unknown rather than any

Reply:
I'm not sure there is a class to check... I'll look.

Reply:
The class is just `Error`, so there's nothing good to instanceof

Reply:
Oh... yeah meant that the catch() argument might not be an Error. Done.